File: JavaTest/CollectionsTest.java
import MyLibrary.*;
import org.junit.*;

import static org.junit.Assert.*;

public class CollectionsTest {
    @Test
    public void VectorPassingTest() {
        StringVector v = new StringVector();
        v.add("a");
        v.add("b");
        v.add("c");

        assertEquals(3, CollectionUserExample.GetCount(v));
        assertEquals("b", CollectionUserExample.GetObjectAtIndex(v, 1));
    }

    @Test
    public void MapUseTest() {
        StringToIntMap m = new StringToIntMap();
        m.set("1", 1);
        m.set("2", 2);
        m.set("3", 3);

        assertEquals(3, m.size());
        assertEquals(3, m.get("3"));
    }
}

File: CSharpTest/CollectionsTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CSharpTest
{
    [TestClass]
    public class CollectionsTest
    {
        [TestMethod]
        public void VectorPassingTest()
        {
            var v = new MyLibrary.StringVector();
            v.Add("a");
            v.Add("b");
            v.Add("c");

            Assert.AreEqual(3, MyLibrary.CollectionUserExample.GetCount(v));
            Assert.AreEqual("b", MyLibrary.CollectionUserExample.GetObjectAtIndex(v, 1));
        }

        [TestMethod]
        public void MapUseTest()
        {
            var m = new MyLibrary.StringToIntMap();
            m["1"] = 1;
            m["2"] = 2;
            m["3"] = 3;

            Assert.AreEqual(3, m.Count);
            Assert.AreEqual(3, m["3"]);
        }
    }
}
